syntax.qmd

```diff
@@ -364,41 +364,56 @@ while (waiting_for_something()) {
 
 ### If statements
 
-* When present, `else` should be on the same line as `}`.
+* A single line if statement must never contain braced expressions. You can use
+  single line if statements for very simple statements that don't have
+  side-effects and don't modify the control flow.
 
-* `&` and `|` should never be used inside of an `if` clause because they can
-  return vectors. Always use `&&` and `||` instead.
+  ```{r}
+  # Good
+  message <- if (x > 10) "big" else "small"
 
-* NB: `ifelse(x, a, b)` is not a drop-in replacement for `if (x) a else b`.
-  `ifelse()` is vectorised (i.e. if `length(x) > 1`, then `a` and `b`  will be
-  recycled to match) and it is eager (i.e. both `a` and `b` will always be
-  evaluated).
+  # Bad
+  message <- if (x > 10) { "big" } else { "small" }
+  ```
 
-  If you want to rewrite a simple but lengthy `if` block:
+* A multiline if statement must contain braced expressions.
 
   ```{r}
+  # Good
+  if (x > 10) {
+    x * 2
+  }
+
   if (x > 10) {
-    message <- "big"
+    x * 2
   } else {
-    message <- "small"
+    x * 3
   }
-  ```
 
-  Just write it all on one line:
+  # Bad
+  if (x > 10)
+    x * 2
 
-  ```{r}
-  message <- if (x > 10) "big" else "small"
+  # In particular, the outer braces are required for this to even parse.
+  {
+    if (x > 10)
+      x * 2
+    else
+      x * 3
+  }
   ```
 
-### Inline statements {#inline-statements}
+* When present, `else` should be on the same line as `}`.
 
-It's ok to drop the curly braces for very simple statements that fit on one line, as long as they don't have side-effects.
+* `&` and `|` should never be used inside of an `if` clause because they can
+  return vectors. Always use `&&` and `||` instead.
 
-```{r}
-# Good
-y <- 10
-x <- if (y < 20) "Too low" else "Too high"
-```
+* NB: `ifelse(x, a, b)` is not a drop-in replacement for `if (x) a else b`.
```

Comment:
Might  be better to include this and the previous bullet in a `::: {.callout-note}` since they're more general programming advice than style?